### ISSUE TITLE
Intermittent failure of agent spec

### ISSUE THREAD

**Issue by andrewvc** [elastic/logstash#5445]
I got  an error on https://travis-ci.org/elastic/logstash/builds/135021910

```

  1) LogStash::Agent#execute when auto_reload is true if state is clean should periodically reload_state
     Failure/Error: expect(subject).to receive(:reload_state!).at_least(3).times
       (#<LogStash::Agent:0x250268ef>).reload_state!(any args)
           expected: at least 3 times with any arguments
           received: 1 time with any arguments
     # ./logstash-core/spec/logstash/agent_spec.rb:158:in `(root)'
     # ./vendor/bundle/jruby/1.9/gems/rspec-wait-0.0.8/lib/rspec/wait.rb:46:in `(root)'
     # ./rakelib/test.rake:42:in `(root)'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/task.rb:248:in `execute'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/task.rb:243:in `execute'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/task.rb:187:in `invoke_with_call_chain'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/task.rb:180:in `invoke_with_call_chain'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/task.rb:173:in `invoke'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/application.rb:150:in `invoke_task'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/application.rb:106:in `top_level'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/application.rb:106:in `top_level'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/application.rb:115:in `run_with_threads'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/application.rb:100:in `top_level'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/application.rb:78:in `run'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/application.rb:176:in `standard_exception_handling'
     # ./vendor/bundle/jruby/1.9/gems/rake-11.1.2/lib/rake/application.rb:75:in `run'
```

**Comment by purbon** [comment]
this was hit again in https://logstash-ci.elastic.co/job/elastic+logstash+master+multijob-intake/196/console

**Comment by purbon** [comment]
this was hit again https://logstash-ci.elastic.co/job/elastic+logstash+master+multijob-os-compatibility/os=debian/134/console

**Comment by jsvd** [comment]
should be fixed by #6039
